lib/say_when/scheduler.rb

```diff
@@ -68,42 +68,69 @@ def start
 
         logger.info "SayWhen::Scheduler running"
         job = nil
+        reset_next_at = Time.now
         while running
           begin
             time_now = Time.now
-            logger.debug "SayWhen:: Looking for job that should be ready to fire before #{time_now}"
-            job = job_class.acquire_next(time_now)
+
+            if reset_acquired_length > 0 && reset_next_at <= time_now
+              reset_next_at = time_now + reset_acquired_length
+              logger.debug "SayWhen:: reset acquired at #{time_now}, try again at #{reset_next_at}"
+              job_class.reset_acquired(reset_acquired_length)
+            end
+
+            begin
+              logger.debug "SayWhen:: Looking for job that should be ready to fire before #{time_now}"
+              job = job_class.acquire_next(time_now)
+            rescue StandardError => ex
+              job_error("Failure to acquire job", job, ex)
+              job = nil
+            end
+
             if job.nil?
               logger.debug "SayWhen:: no jobs to acquire, sleep"
               sleep(tick_length)
-            else
+              next
+            end
+
+            begin
               logger.debug "SayWhen:: got a job: #{job.inspect}"
               # delegate processing the trigger to the processor
               self.processor.process(job)
               logger.debug "SayWhen:: job processed"
 
-              # this should update next fire at, and put back in list of scheduled jobs
+              # if successful, update next fire at, put back to waiting / ended
               job.fired(time_now)
               logger.debug "SayWhen:: job fired complete"
+            rescue StandardError=>ex
+              job_error("Failure to process", job, ex)
             end
-          rescue Interrupt
-            job_msg = job && " job:'#{job.inspect}'"
-            logger.error "\nSayWhen:: Interrupt! #{job_msg}"
-            exit
-          rescue StandardError=>ex
-            job_msg = job && " job:'#{job.inspect}'"
-            logger.error "SayWhen:: Failure to process#{job_msg}: #{ex.message}\n\t#{ex.backtrace.join("\t\n")}"
-            job.release if job
-          rescue Exception=>ex
-            logger.error "SayWhen:: Exception in process#{job_msg}: #{ex.message}\n\t#{ex.backtrace.join("\t\n")}"
-            exit
+
+          rescue Interrupt => ex
+            job_error("Interrupt!", job, ex)
+            raise ex
+          rescue StandardError => ex
```

Comment:
correct, and with the database.yml set to reconnect, it will then eventually reconnect without needing to restart the process.

If you don't have that connection setting, then it will not reconnect.

I suppose another way to do this would be to add that reconnect logic myself, rather than rely on rails config.